plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -365,10 +365,14 @@ else if (type instanceof ScaledFloatType) {
             return new TypeAndDecoder(DOUBLE, new DoubleDecoder.Descriptor(path));
         }
         else if (type instanceof DateTimeType) {
-            if (((DateTimeType) type).getFormats().isEmpty()) {
-                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path));
+            List<String> formats = ((DateTimeType) type).getFormats();
+            if (formats.isEmpty()) {
+                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path, null));
+            }
+            else {
+                String patterns = String.join("||", formats);
+                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path, patterns));
```

Comment:
Can make this more concise like?

```suggestion
            String patterns = null;
            if (!formats.isEmpty()) {
                patterns = String.join("||", formats);
            }
            return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path, patterns));
```